File: matlab/transpiled_matlab/reverse_main.py
import numpy
from PIL import Image

source = numpy.asarray(Image.open('../images/source.jpg'))

block_widths = [2, 4, 8, 16, 32, 64, 128, 256, 512]

for block_width in block_widths:
    filename = "../results/p-" + str(512 / block_width) + ".bmp"
    image = numpy.asarray(Image.open(filename))
    
    restored_img = reverse_perturbation(image)
    equal = numpy.allclose(source, restored_img)
    #display if two are equal
    print(equal)

File: matlab/reverse_main.m
clc, clear

source = imread('../images/source.jpg');

for block_width =[2, 4, 8, 16, 32, 64, 128, 256]

    filename = ['../results/p-', int2str(512/block_width), '.bmp'];
    image = imread(filename);
   
    restored_img = reverse_perturbation(image);
    
    % Print to confirm restored image matches source
    all(source == restored_img, 'all')
end